tests/common.go

```diff
@@ -3779,6 +3771,148 @@ func CreateClusterPairFile(pairInfo map[string]string, skipStorage, resetConfig
 	return addStorageOptions(pairInfo, clusterPairFileName)
 }
 
+func ScheduleBidirectionalClusterPair(cpName, cpNamespace, projectMappings string, objectStoreType storkv1.BackupLocationType, secretName string, mode string, sourceCluster int, destCluster int) error {
+	//var token string
+	// Setting kubeconfig to source because we will create bidirectional cluster pair based on source as reference
+	err := SetCustomKubeConfig(sourceCluster)
+	if err != nil {
+		return err
+	}
+
+	// Create namespace for the cluster pair on source cluster
+	_, err = core.Instance().CreateNamespace(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cpNamespace,
+			Labels: map[string]string{
+				"creator": "stork-test",
+			},
+		},
+	})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failed to create namespace %s on source cluster", cpNamespace)
+	}
+
+	srcKubeConfigPath, err := GetCustomClusterConfigPath(sourceCluster)
+
+	defer func() error {
+		config, err := clientcmd.BuildConfigFromFlags("", srcKubeConfigPath)
+		if err != nil {
+			return err
+		}
+		core.Instance().SetConfig(config)
+		apps.Instance().SetConfig(config)
+		stork.Instance().SetConfig(config)
+		return nil
+	}()
+
+	err = SetCustomKubeConfig(destCluster)
+	if err != nil {
+		return err
+	}
+
+	// Create namespace for the cluster pair on destination cluster
+	_, err = core.Instance().CreateNamespace(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cpNamespace,
+			Labels: map[string]string{
+				"creator": "stork-test",
+			},
+		},
+	})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failed to create namespace %s on destination cluster", cpNamespace)
+	}
+
+	destKubeConfigPath, err := GetCustomClusterConfigPath(destCluster)
+
+	err = SetCustomKubeConfig(sourceCluster)
+	if err != nil {
+		return err
+	}
+
+	// Create source --> destination and destination --> cluster pairs using storkctl
+	factory := storkctl.NewFactory()
+	cmd := storkctl.NewCommand(factory, os.Stdin, os.Stdout, os.Stderr)
+	cmdArgs := []string{"create", "clusterpair", "-n", cpNamespace, cpName,
+	    "--kubeconfig", srcKubeConfigPath,
+		"--src-kube-file", srcKubeConfigPath,
+		"--dest-kube-file", destKubeConfigPath,
+	}
+
+	if mode == "sync-dr" {
+		cmdArgs = []string{"create", "clusterpair", "-n", cpNamespace, cpName,
+		    "--kubeconfig", srcKubeConfigPath,
+			"--src-kube-file", srcKubeConfigPath,
+			"--dest-kube-file", destKubeConfigPath,
+			"--mode", "sync-dr",
+	    }
+	}
+
+	if projectMappings != "" {
+		cmdArgs = append(cmdArgs, "--project-mappings")
+		cmdArgs = append(cmdArgs, projectMappings)
```

Comment:
For brevity

```go
cmdArgs = append(cmdArgs, "--project-mappings", projectMappings)
```

Reply:
We tried this and it fails sometimes with some space left out, so above one is the existing way which I used.